### DISCUSSION TITLE
Seeking Clarity: Code Coverage Implementation with Gradle

### DISCUSSION THREAD

**Discussion by Rd4dev** [oppia/oppia-android#5374] [GSoC 2024] [answered]
I'm seeking clarification regarding the implementation of code coverage instrumentation. The [project description](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2024#41-code-coverage-support-and-enforcement) highlights issues with generating accurate reports using Gradle and mentions transitioning away from Gradle to Bazel.

> Past attempts (see [#2466](https://github.com/oppia/oppia-android/pull/2466) and [#2744](https://github.com/oppia/oppia-android/pull/2744)) ran into challenges with generating correct reports using Gradle.

Would appreciate clarification on whether the project will primarily focus on integrating code coverage instrumentation with Bazel alone, or if it will also extend support for Gradle.

Attempted to run the coverage with Jacoco 0.7.9 using CLI, and below are the results.

**Domain module:**
![1](https://github.com/oppia/oppia-android/assets/122200035/69549877-8792-43ee-96bf-fc3185ae68ea)

**AudioPlayerController:**
![2](https://github.com/oppia/oppia-android/assets/122200035/6d925d3a-1b2e-4ae4-8a0a-09ff248840cb)

**File Report:**
![3](https://github.com/oppia/oppia-android/assets/122200035/b3f9ef26-d49d-4e26-9e10-6dc8d47a4b3c)

**Test Cases:**
![4](https://github.com/oppia/oppia-android/assets/122200035/55b0a54f-4cc4-4d50-8113-bf42265f57f2)

**Comment by adhiamboperes** [comment, marked as answer]
The focus of the project should be bazel, because we are soon migrating awy from gradle.

**Reply by Rd4dev** [reply]
Thanks for the clarification @adhiamboperes 👍
